### DISCUSSION TITLE
jruby installation failing on latest macOS - no relevant logging

### DISCUSSION THREAD

**Discussion by aarestad** [rbenv/ruby-build#2403] [Build failures] [answered]
```
$ rbenv install --verbose jruby-9.4.7.0
cd /var/folders/0y/907n_q_x339dwbg5m40smsz40000gn/T/ruby-build.20240604120233.49388.Nuv2qd
==> Downloading jruby-dist-9.4.7.0-bin.tar.gz...
-> curl -q -fL -o jruby-9.4.7.0.tar.gz https://dqw8nmjcqpjn7.cloudfront.net/f1c39f8257505300a528ff83fe4721fbe61a855abb25e3d27d52d43ac97a4d80
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 30.3M  100 30.3M    0     0  9803k      0  0:00:03  0:00:03 --:--:-- 9806k
cd /var/folders/0y/907n_q_x339dwbg5m40smsz40000gn/T/ruby-build.20240604120233.49388.Nuv2qd/jruby-9.4.7.0
==> Installing jruby-9.4.7.0...

BUILD FAILED (macOS 14.5 on arm64 using ruby-build 20240530.1)

You can inspect the build directory at /var/folders/0y/907n_q_x339dwbg5m40smsz40000gn/T/ruby-build.20240604120233.49388.Nuv2qd

BUILD FAILED (macOS 14.5 on arm64 using ruby-build 20240530.1)

You can inspect the build directory at /var/folders/0y/907n_q_x339dwbg5m40smsz40000gn/T/ruby-build.20240604120233.49388.Nuv2qd

$  ls -l /var/folders/0y/907n_q_x339dwbg5m40smsz40000gn/T/ruby-build.20240604120233.49388.Nuv2qd/jruby-9.4.7.0 
total 176
-rw-r--r--   1 aarestad  staff   1282 Apr 29 13:24 BSDL
-rw-r--r--   1 aarestad  staff  76023 Apr 29 13:24 COPYING
-rw-r--r--   1 aarestad  staff    218 Apr 29 13:24 LEGAL
-rw-r--r--   1 aarestad  staff   2581 Apr 29 13:24 LICENSE.RUBY
drwxr-xr-x  44 aarestad  staff   1408 Apr 29 13:26 bin
drwxr-xr-x   5 aarestad  staff    160 Jun  4 12:02 lib
drwxr-xr-x  16 aarestad  staff    512 Jun  4 12:02 samples
```

The log gave no indication of what actually failed, and it looks like it is expanding the jruby package in the temp directory. I'm flummoxed. :(

**Comment by mislav** [comment, marked as answer]
Hi, sorry for the trouble! I suspect that this line failed:

https://github.com/rbenv/ruby-build/blob/076d1a23169bb715942448d368be42424809d1e6/bin/ruby-build#L873

When it fails, ruby-build would abort but the error stream is unfortunately silenced, so nothing would get printed in ruby-build output. I've opened a PR to avoid/fix this.

To get to the bottom of why this fails for you, you could try cd'ing into your ruby-build temporary directory and running `jruby-9.4.7.0/bin/jruby -e "puts JRUBY_VERSION"` there.

**Comment by aarestad** [comment]
Weirdly, I just tried again and it worked? 🤔 I'll leave this open in case someone else hits a similar issue.

**Reply by mislav** [reply]
Did you upgrade ruby-build between attempts? I had merged https://github.com/rbenv/ruby-build/issues/2406 based on your report.

**Comment by aarestad** [comment]
I don't think so, but it's possible I did in passing. :) Anyway, It Just Works. Now I have to track down a zillion other JRuby issues with my huge monolith app. 🫠 Thank you for your help!

**Reply by mislav** [reply]
Happy it works now. I'm going to mark my best guess as an answer, and since the aforementioned PR shipped, there is a good chance that whatever caused the failure will be logged in the future. So here's to hoping no one will need to scratch their head over this.
